NitroxClient/GameLogic/Cyclops.cs

```diff
@@ -1,15 +1,37 @@
-﻿using NitroxClient.Communication.Abstract;
+﻿using System.Collections.Generic;
+using System.Linq;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic.Helper;
+using NitroxClient.Unity.Helper;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.DataStructures.Util;
+using NitroxModel.Helper;
+using NitroxModel.Logger;
 using NitroxModel.Packets;
+using UnityEngine;
 
 namespace NitroxClient.GameLogic
 {
     public class Cyclops
     {
         private readonly IPacketSender packetSender;
+        private readonly SimulationOwnership simulationOwnershipManager;
 
-        public Cyclops(IPacketSender packetSender)
+        /// <summary>
+        /// KeyValuePair<Guid, douseAmount>. Used in <see cref="OnFireDoused(Fire, SubRoot, float)"/> to reduce
+        /// the <see cref="CyclopsFireHealthChanged"/> packet spam as fires are being doused. A packet is only sent after
+        /// the douse amount surpasses <see cref="FIRE_DOUSE_AMOUNT_TRIGGER"/>
+        /// </summary>
+        private readonly Dictionary<string, float> fireDouseAmount = new Dictionary<string, float>();
```

Comment:
Should we have a helper class to separate out all of the first related logic?  There is a ton of it and that would allow for some smaller method sizes when refactoring.

Reply:
At least some of the fire logic will have to be refactored out of Cyclops at some point. I haven't taken a close look, but it seems like the Fire class is pretty much the only class that the Cyclops code shares with the rest of the code base.

Despite that, it is still wrapped in it's own special damage system via SubFire that manages fire creation and storage. It seems like the only thing it shares is extinguishing.

I can't seem to find any dynamic fire creation beyond the Cyclops. I suspect there's no other dynamic fires, and the ones in the Aurora are all static and unable to be respawned. Either way, I need to modify Fire_Douse_Patch to allow for fires that are not in a Cyclops, and rename CyclopsFireHealthChangedProcessor to FireHealthChangedProcessor and allow for non-Cyclops fires as well.